### PR TITLE
History Error Injection Client

### PR DESCRIPTION
**Pull request by yycptt**
**What changed?**
Add an error injection client for history client

**Why?**
Injection random client error during in bench test to expose potential error handling bugs.

**How did you test it?**
Local test.

**Potential risks**
No risk. Error injection client won't be created if error rate is 0 on service start up.

### CODE REVIEW

File: service/worker/parentclosepolicy/workflow.go [hunk 1]
```diff
@@ -24,6 +24,8 @@ import (
 	"context"
 	"time"
 
+	"github.com/uber/cadence/common/types/mapper/thrift"
```
Comment (vytautas-karpavicius) [inline service/worker/parentclosepolicy/workflow.go:27]: nit: this seems like it is not used anywhere

**Review by vytautas-karpavicius — APPROVED** [review]
I know this is a common trend in this repo to add such decorators. But did anyone considered to use code generation for such occasions? This seems like it could be easily generated from a template.

File: service/worker/parentclosepolicy/workflow.go (hunk 1 above)
Comment (yycptt) [reply service/worker/parentclosepolicy/workflow.go]: Good catch. Removed.

**Comment by yycptt** [comment]
> I know this is a common trend in this repo to add such decorators. But did anyone considered to use code generation for such occasions? This seems like it could be easily generated from a template.

I think that's a good idea, let me create an issue to track it.

**Pull request merged**